astropy/nddata/nduncertainty.py

```diff
@@ -186,6 +186,24 @@ def __repr__(self):
             body = str(self.array)
         return ''.join([prefix, body, ')'])
 
+    def __getstate__(self):
+        # Because of the weak reference the class wouldn't be picklable.
+        try:
+            return self._array, self._unit, self.parent_nddata
+        except MissingDataAssociationException:
+            # In case there's no parent
+            return self._array, self._unit
```

Comment:
Nitpick : I would also return a 3rd item with `None` here, which would make the `__setstate__` simpler with always 3 items.

Reply:
Good idea. 👍